packages/junto-utils/react/ChatContext.tsx

```diff
@@ -195,125 +254,61 @@ export function ChatProvider({ perspectiveUuid, children, channelId }: any) {
 
   async function handleLinkAdded(link) {
     console.log("Got message link", link);
+
     const agent = await getMe();
 
-    if (linkIs.message(link)) {
-      const isSameChannel = await ad4mClient.perspective.queryProlog(perspectiveUuid, `triple("${channelId}", "${DIRECTLY_SUCCEEDED_BY}", "${link.data.target}").`);
-      if (isSameChannel) {
-        const message = getMessage(link);
+    const isMessageFromSelf = link.author === agent.did;
+    if (!isMessageFromSelf) {
+      if (linkIs.message(link)) {
+        const isSameChannel = await ad4mClient.perspective.queryProlog(perspectiveUuid, `triple("${channelId}", "${DIRECTLY_SUCCEEDED_BY}", "${link.data.target}").`);
+        if (isSameChannel) {
+          const message = getMessage(link);
+  
+          if (message) {
+            setState((oldState) => addMessage(oldState, message));
+  
+            setState((oldState) => ({
+              ...oldState,
+              isMessageFromSelf: false,
+            }));
+          }
+        }
+      }
 
-        if (message) {
-          setState((oldState) => addMessage(oldState, message));
+      if (linkIs.reaction(link)) {
+        addReactionToState(link);
+      }
 
+      if (linkIs.reply(link)) {
+        const isSameChannel = await ad4mClient.perspective.queryProlog(perspectiveUuid, `triple("${channelId}", "${DIRECTLY_SUCCEEDED_BY}", "${link.data.source}").`);
+  
+        if (isSameChannel) {
+          const message = getMessage(link);
+  
+          setState((oldState) => addMessage(oldState, message));
+    
           setState((oldState) => ({
             ...oldState,
-            isMessageFromSelf: link.author === agent.did,
+            isMessageFromSelf: false,
```

Comment:
should the replies and reactions not also check if message is from self?

Reply:
They are in a single if-condition